src/calliope/config/model_data_checks.yaml

```diff
@@ -29,6 +29,15 @@ fail:
   - where: storage_initial>1
     message: "storage_initial is a fraction; values larger than 1 are not allowed."
 
+  - where: storage_initial<0
+    message: "storage_initial is a fraction; values smaller than 0 are not allowed."
+
+  - where: flow_cap_min<0
```

Comment:
tbh, the choice of what to catch here is increasingly difficult to justify. You could set this kind of requirement on pretty much every parameter. A better approach would be to extract the min/max values from the schema and to use those once the data is loaded in.

Reply:
I'm not sure if this is the most efficient way to approach the problem, but the underlying idea would be: Parameter checks should catch all errors that may otherwise lead to something that a user possibly cannot easily understand. That is either something that (1) leads to an obscure internal error (e.g., a non-existing key in some lookup that is not directly related to the initial mistake), or (2) leads to an outcome that is hard to track back to the initial mistake.

Related to (2), an example, setting:

```yaml
storage_cap_max: 3000
storage_cap_min: 3001
```

for a tech, results in a `BackendWarning: Model solution was non-optimal.` (due to the infeasibility). Of course it's evident here, but for larger models, that may be generated from something, or based on parameters/config from somewhere this could be really hard to find (if one does not have access to an IIS calculation).

Reply:
This race to the bottom with data checking caused us quite some maintenance burden in v0.6, hence why we have removed it in favour of the user taking on more responsibility. We can't hope to capture _everything_ that they might do to create an infeasible model, especially once they add their own parameters and math. Even without their own math, there are complex combinations of pre-defined parameters that could lead to infeasibility (setting an resource_area_min, linking area and flow_cap, then setting a flow_cap_max that violates that, for instance).

Reply:
Perfectly fine, makes sense - just wanted to point out these immediate relations. I'll remove these from the PR, since they would otherwise just be inconsistent with other parameters, and the `< 0` does not matter anyways ...